src/components/Api.js

```diff
@@ -0,0 +1,146 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, {
+      headers: this._headers,
+    })
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        } else {
+          return Promise.reject(`Unexpected Error: ${res.status}`);
+        }
+      })
+      .catch((err) => {
+        console.error(err); // log the error to the console
+      });
+  }
+
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: this._headers,
+    }).then((response) => response.json()); // Parse the response as JSON
```

Comment:
` .then(this._checkResponse)` should be everywhere to check if the response is ok or not